### PR TITLE
Address some comments

### PR DESCRIPTION
**Pull request by guozhangwang**
@ymatsuda

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/KafkaStreaming.java [hunk 1]
```diff
@@ -68,12 +71,12 @@
     private static final int STOPPED = 2;
     private int state = CREATED;
 
-    private final StreamThread[] threads;
+    private final List<StreamThread> threads;
```
Comment (ymatsuda) [inline streams/src/main/java/org/apache/kafka/streams/KafkaStreaming.java]: What is wrong with the array?

File: streams/src/test/java/org/apache/kafka/test/MockProcessorDef.java [hunk 2]
```diff
@@ -22,11 +22,12 @@
 import org.apache.kafka.streams.processor.ProcessorDef;
 
 import java.util.ArrayList;
+import java.util.List;
 
 public class MockProcessorDef<K, V> implements ProcessorDef {
 
-    public final ArrayList<String> processed = new ArrayList<>();
-    public final ArrayList<Long> punctuated = new ArrayList<>();
+    public final List<String> processed = new ArrayList<>();
```
Comment (ymatsuda) [inline streams/src/test/java/org/apache/kafka/test/MockProcessorDef.java]: ArrayList is fine. No need to use List.

File: streams/src/test/java/org/apache/kafka/test/MockProcessorDef.java [hunk 3]
```diff
@@ -22,11 +22,12 @@
...
+    public final List<Long> punctuated = new ArrayList<>();
```
Comment (ymatsuda) [inline streams/src/test/java/org/apache/kafka/test/MockProcessorDef.java]: Same here.

File: streams/src/test/java/org/apache/kafka/test/MockProcessorContext.java [hunk 4]
```diff
@@ -37,7 +38,7 @@
     private Serializer serializer;
     private Deserializer deserializer;
     private ProcessorNode node;
-    private final ArrayDeque<ProcessorNode> nodeStack = new ArrayDeque<ProcessorNode>();
+    private final Deque<ProcessorNode> nodeStack = new ArrayDeque<ProcessorNode>();
```
Comment (ymatsuda) [inline streams/src/test/java/org/apache/kafka/test/MockProcessorContext.java]: ArrayDeque is fine. No need to declare the variable as Deque interface

File: streams/src/main/java/org/apache/kafka/streams/KafkaStreaming.java (hunk 1 above)
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/KafkaStreaming.java]: I was originally thinking arrays should be avoided in general, also suggested by someone from OS reviews:

http://eclipsesource.com/blogs/2014/04/11/3-good-reasons-to-avoid-arrays-in-java-interfaces/

But thinking of this a bit more now I feel this is fine in our case here. Will change it back.

Comment (ymatsuda) [inline streams/src/main/java/org/apache/kafka/streams/KafkaStreaming.java]: The article is talking about the use of array as an interface. Here we are using the array as an internal structure.

**Comment by ymatsuda** [comment]
lgtm

**Pull request merged**